auraed/src/runtime/cells/cell.rs

```diff
@@ -129,3 +189,48 @@ fn hierarchy() -> Box<dyn Hierarchy> {
     // hierarchies::V2
     Box::new(hierarchies::V2::new())
 }
+
+#[derive(Error, Debug)]
+pub(crate) enum CellError {
+    #[error("cell '{cell_name}' already exists'")]
+    Exists { cell_name: CellName },
+    #[error("cell '{cell_name}' not found'")]
+    NotFound { cell_name: CellName },
+    #[error("cell '{cell_name}' could not be freed: {source}")]
+    FailedToFree { cell_name: CellName, source: cgroups_rs::error::Error },
+    #[error(
+        "cell '{cell_name}' already has an executable '{executable_name}'"
+    )]
+    ExecutableExists { cell_name: CellName, executable_name: ExecutableName },
+    #[error("cell '{cell_name} could not find executable '{executable_name}'")]
+    ExecutableNotFound { cell_name: CellName, executable_name: ExecutableName },
+    #[error("cell '{cell_name}': {source}")]
+    ExecutableError { cell_name: CellName, source: ExecutableError },
+    #[error("cell '{cell_name}' failed to add executable (executable:?)")]
+    FailedToAddExecutable {
+        cell_name: CellName,
+        executable: Executable,
+        source: cgroups_rs::error::Error,
+    },
+}
+
+impl From<CellError> for Status {
+    fn from(err: CellError) -> Self {
+        let msg = err.to_string();
+        error!("{msg}");
+        match err {
+            CellError::Exists { .. } | CellError::ExecutableExists { .. } => {
+                Status::already_exists(msg)
+            }
+            CellError::NotFound { .. }
+            | CellError::ExecutableNotFound { .. } => Status::not_found(msg),
+            // TODO (future-highway): I don't know what the conventions are of revealing
+            //  messages that reveal the workings of the system to the api consumer
+            //  in this type of application.
+            //  For now, taking the safe route and not exposing the error messages for the below errors.
```

Comment:
this won't expose it to the client, but the auraed will still log it above (l:220).  up to you how much you want to provide to the client vs only have in the server logs :)

Reply:
I'm not sure my comment was clear enough. So...

For example, the `CellError::FailedToFree` variant has a `source: cgroups_rs::error::Error` field and an error template of `"cell '{cell_name}' could not be freed: {source}"`, so source ends up in the error message, which I was concerned about ending up in the `Status`'s message that is returned from `CellService`.

Reply:
sure.. is it going to be the same person seeing the Status Response as is running auraed (in which it's logged to stderr)?

Reply:
Sounds like you're saying it's fine to expose, which makes perfect sense considering the observability api. I'm still training my brain to ignore the lessons ingrained from writing other programs.